### DISCUSSION TITLE
[Feature Request] Wider Log-Viewer/Terminal

### DISCUSSION THREAD

**Discussion by vinnpinn** [louislam/dockge#31] [Feature Request]
Currently most log entries are broken into multiple lines since the Log Viewer ist quite small, making it quite hard to read:
![image](https://github.com/louislam/dockge/assets/26883610/3f816414-6e5d-42da-b0ee-545b96e8d790)

Maybe moving the Terminal above Containers and compose.yaml might work, just something I quickly threw together using my limited inspect element knowledge:
![Screenshot 2023-11-13 123839](https://github.com/louislam/dockge/assets/26883610/6c8b957a-ffe0-4f57-baee-b8c8738e09a0)

**Comment by louislam** [comment]
Off topic, just want to let you know, since the volumes config is not correct, some data will be stored in `/my-stacks` in the host

The correct one should be:
```
volumes:
  ...
  - /docker:/docker
environment:
  - DOCKGE_STACKS_DIR=/docker
```

Don't forget to put `/my-stacks`'s data back to `/docker`.

**Reply by holocronology** [reply]
Took me a few minutes to figure out this is what you meant in your notes. Maybe it needs to be clearer in the examples:
![image](https://github.com/louislam/dockge/assets/65265114/204bafe2-3d07-4b46-a60b-f56c7fa3b99f)

**Reply by vinnpinn** [reply]
oh, thank you :)

**Reply by holocronology** [reply]
Also, you refer to /my-stacks and /opt/stacks in different examples.

This is what I ended up with:
![image](https://github.com/louislam/dockge/assets/65265114/bfc78053-26b2-4ced-b11a-fbdbef263620)

**Reply by louislam** [reply]
@holocronology You cannot do that, as I said. Any paths that written in relative path (e.g. `./data`) will be written into `/opt/stacks/<stackName>/data` in your case.

**Reply by holocronology** [reply]
Are you suggesting my above compose shouldn't work? It does.

What should it be?

**Comment by holocronology** [comment]
The compose.yaml box suffers from similar width issues, word/line wrapping, etc.

It would be better in a wider section also.

**Reply by trythatagain** [reply]
Perhaps when editing it uses most of the width by moving the 'Network box' along side container box and then having the edit widow below that and use all the width?

**Comment by Emiliaaah** [comment]
This feature would be great to have, maybe even an optional full screen option? For when you have to dive into the logs.

**Comment by mdeeter** [comment]
I too wish I could make the terminal wider to be easier to read output and editor bigger when editing.

I'd be happy if the instance list on the left was collapsible.... or at least narrower... my instance names generally cover 50% of the width of the left panel.

Alternatively, another change that could make it easier to view would be to just move the `compose.yml` editor to the bottom of the page.

Even better would be to create a grid system and let the user select a layout where they could move the panels to whatever order they like.


Currently, I'm making it bearable by zooming the page out to 80%... but then the left instance-list panel becomes even more wasteful.

**Comment by eded333** [comment]
https://github.com/louislam/dockge/discussions/437#discussioncomment-8732500

**Comment by draeron** [comment]
compose and logs frame should be in the same column aka one over the other to be full width, use a tab or accordion

**Comment by mdeeter** [comment]
I ended up using uBlock with Chome to make mine like this:
![image](https://github.com/user-attachments/assets/84640405-a4e4-4be5-9e73-3f6fa16e150d)

 using these uBlock settings to insert CSS
![image](https://github.com/user-attachments/assets/525eb66c-697a-4cfa-8123-8fea85121ece)

**Reply by aleeraser** [reply]
@mdeeter rules for easier copying&pasting

```
DOCKGE_URL##div.col-lg-6:style(max-width: none !important; width: 100% !important;)
DOCKGE_URL##div.terminal:style(font-size:0.6em !important;)
DOCKGE_URL##div.container.mb-3.big-padding.shadow-box:style(max-width: none !important; width: 100% !important) 
```

**Reply by zanphear** [reply]
YOU Sir are an absolute legend. I didn't even know uBlock could do this. This has opened up an entire playground for me I didn't know existed!

**Reply by ReanimationXP** [reply]
+1, legendary. THANK YOU!

p.s. OP, @aleeraser, and for anyone who wants to use this, and OP, obviously replace `DOCKGE_URL` with your instance domain or IP.  Note that it should be **ONLY your domain OR the IP**.. despite the name do not enter the full URL including http and such. For example,

```
domain.com##div.col-lg-6:style(max-width: none !important; width: 100% !important;)
domain.com##div.terminal:style(font-size:0.6em !important;)
domain.com##div.container.mb-3.big-padding.shadow-box:style(max-width: none !important; width: 100% !important)
```

**Reply by molivier-io** [reply]
I was hesitating to keep dockge because if this readability issue. But now it feels my expectations. Thank you!

**Reply by Festival0103** [reply]
For those running multiple instances where you can't specify a port: I found that you can list them separated by commas and then add a text filter in the body (using 'Dockge' as an example). I’ve checked other places where the word 'Dockge' might appear, and so far so good, but you should double-check this for your own setup

```
dockge.url,192.168.1.2,192.168.1.3##body:has-text(Dockge) div.col-lg-6:style(max-width: none !important; width: 100% !important;)
dockge.url,192.168.1.2,192.168.1.3##body:has-text(Dockge) div.terminal:style(font-size:0.6em !important;)
dockge.url,192.168.1.2,192.168.1.3##body:has-text(Dockge) div.container.mb-3.big-padding.shadow-box:style(max-width: none !important; width: 100% !important)
```

**Comment by pyxl-dev** [comment]
Hello ! I've worked on an expand panel for your feature request.

Here is the PR : https://github.com/louislam/dockge/pull/678

**Comment by revenz** [comment]
Heres the style just as css if you want to use a custom css extension

```
div.col-lg-6 {
    max-width: none !important;
    width: 100% !important;
}

div.termina {
    font-size: 0.6em !important;
}

div.container.mb-3.big-padding.shadow-box {
    max-width: none !important;
    width: 100% !important
}

```

**Reply by ruruchiii** [reply]
Just what I was looking for, thanks.

**Reply by ReanimationXP** [reply]
I always forget I can do this.. lol.

**Comment by memilanuk** [comment]
Seems to be a pretty common/popular request... any chance this is going to get rolled into the main release sometime soon?  Not everybody is working on a big widescreen monitor - those of us on laptops are suffering a bit here ;)

**Reply by ReanimationXP** [reply]
I'll submit a pull request for this soon but idk how frequently Dockge is maintained anymore, it hasn't had an update since May and has 39 open pull requests which is a pretty insane amount for a project of this size tbh.

**Reply by ReanimationXP** [reply]
Actually a commenter above already did it looks like: https://github.com/louislam/dockge/pull/678

**Comment by airtonix** [comment]
having the logs in an entirely different tab all on its own would be more valuable tbh
